pkg/archive/archive.go

```diff
@@ -204,6 +204,35 @@ func DecompressStream(archive io.Reader) (io.ReadCloser, error) {
 	}
 }
 
+// GzipReadSeekerWrapper wraps an io.ReadSeeker for gzip reading
+type GzipReadSeekerWrapper struct {
+	readSeeker io.ReadSeeker
+	*gzip.Reader
+}
+
+// NewGzipReadSeekerWrapper initializes a new GzipReadSeeker
+func NewGzipReadSeekerWrapper(rs io.ReadSeeker) (*GzipReadSeekerWrapper, error) {
+	gzipReader, err := gzip.NewReader(rs)
+	if err != nil {
+		return nil, err
+	}
+	return &GzipReadSeekerWrapper{rs, gzipReader}, nil
+}
+
+// Seek sets the offset for the next Read, and can only seek to the
+// beginning of the file.
+func (g *GzipReadSeekerWrapper) Seek(offset int64, whence int) (int64, error) {
```

Comment:
I don't think this will work. The seeker will need to do more than seek to `0,0`
We'll probably have to decompress to a temp file to make this work.